README.md

```diff
@@ -276,6 +281,38 @@ function App() {
 render(<App />, document.body);
 ```
 
+### Typescript Routes
+
+In some cases, depending on your TypeScript configuration, you may encounter
+an error that prevents you from passing the `path` or `default` props to your components:
+
+```tsx
+error TS2322: Type '{ path: string; myProps: MyProps }' is not assignable to type 'IntrinsicAttributes & MyProps'.
+  Property 'path' does not exist on type 'IntrinsicAttributes & MyProps'.
+
+      <MyComponent path="/component"
+                         ~~~~
+```
+
+In such cases, you can wrap your components in `<Route />`. This will also accept your component's
+props. Keep in mind that both `path` and `default` props are passed down to your component.
+
+```tsx
+import {Router, Route} from 'preact-router';
+
+function App() {
+  let users = getUsers();
+
+  return (
+    <Router>
+      <Route component={Home} path="/" />
+      {/* Route will accept any props of `component` type */}
+      <Route component={Users} users={users} path="/users"  />
+    </Router>
+  );
+}
+```
+
```

Comment:
```suggestion
### `Route` Component

Alternatively to adding the router props (`path`, `default`) directly to your component, you may want to use the `Route` component we provide instead. This tends to appease TypeScript, while still passing down the routing props into your component for use.

```js
import { Router, Route } from 'preact-router';

function App() {
  let users = getUsers();

  return (
    <Router>
      <Route path="/" component={Home} />
      {/* Route will accept any props of `component` type */}
      <Route path="/users" component={Users} users={users}   />
    </Router>
  );
}
```